File: solgrid_tea/blueprints/benchmark.py
from flask import Blueprint, jsonify, request

from solgrid_tea.extensions import db
from solgrid_tea.schemas.benchmark import BenchmarkQuery
from solgrid_tea.security import tenant_scoped
from solgrid_tea.services.benchmark_engine import compute_benchmark

benchmark_bp = Blueprint("benchmark", __name__)


@benchmark_bp.get("")
@tenant_scoped()
def get_benchmark():
    query = BenchmarkQuery.model_validate(
        {
            "facility_id": request.args.get("facility_id"),
            "period_start": request.args.get("period_start"),
            "period_end": request.args.get("period_end"),
        }
    )
    result = compute_benchmark(db.session, query.facility_id, query.period_start, query.period_end)
    return jsonify(result.model_dump(mode="json"))

File: frontend/src/api/benchmark.ts
import { apiRequest } from "./client";
import type { BenchmarkResult } from "../types/api";

export function getBenchmark(params: {
  facility_id: string;
  period_start: string;
  period_end: string;
}): Promise<BenchmarkResult> {
  return apiRequest<BenchmarkResult>("/api/v1/benchmark", { query: params });
}
